src/main/java/net/neoforged/neoforge/client/textures/FluidSpriteCache.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.textures;
+
+import java.util.Map;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.texture.MissingTextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureAtlas;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.BlockAndTintGetter;
+import net.minecraft.world.level.material.FluidState;
+import net.neoforged.neoforge.client.extensions.common.IClientFluidTypeExtensions;
+import org.jetbrains.annotations.ApiStatus;
+
+public final class FluidSpriteCache {
+    private static Map<ResourceLocation, TextureAtlasSprite> textureLookup = Map.of();
+    private static TextureAtlasSprite missingSprite = null;
+
+    public static TextureAtlasSprite[] getFluidSprites(BlockAndTintGetter level, BlockPos pos, FluidState fluid) {
+        IClientFluidTypeExtensions props = IClientFluidTypeExtensions.of(fluid);
+        ResourceLocation overlay = props.getOverlayTexture(fluid, level, pos);
+        Map<ResourceLocation, TextureAtlasSprite> textures = textureLookup;
+
+        return new TextureAtlasSprite[] {
+                textures.getOrDefault(props.getStillTexture(fluid, level, pos), missingSprite),
+                textures.getOrDefault(props.getFlowingTexture(fluid, level, pos), missingSprite),
+                overlay == null ? null : textures.getOrDefault(overlay, missingSprite),
+        };
+    }
+
+    @ApiStatus.Internal
+    @SuppressWarnings("deprecation")
```

Comment:
You can get the sprite from some menu if you want to remove the deprecation. :P

Reply:
Accessing the atlas location via `InventoryMenu.BLOCK_ATLAS` seems like an anti-pattern to me, I'd rather suppress the deprecation warning instead.